### PR TITLE
SoundWire: add ADL topologies

### PR DESCRIPTION
**Pull request by plbossart**
add machine tables based on early hardware information

For tests and checks only for now

### CODE REVIEW

**Comment by kv2019i** [comment]
Thanks @plbossart . I wonder if we should carry these in sof-dev while testing, even if not sending to upstream yet?

File: sound/soc/intel/common/soc-acpi-intel-adl-match.c [hunk 1]
```diff
@@ -177,6 +209,12 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_adl_sdw_machines[] = {
 		.drv_name = "sof_sdw",
 		.sof_tplg_filename = "sof-adl-rt711-rt1316-rt714.tplg",
 	},
+	{
+		.link_mask = 0x5, /* 3 active links required */
```
Comment (RanderWang) [inline sound/soc/intel/common/soc-acpi-intel-adl-match.c]: 2 active links ?

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c]: yes will fix.

**Comment by plbossart** [comment]
> Thanks @plbossart . I wonder if we should carry these in sof-dev while testing, even if not sending to upstream yet?

I wanted these topologies to be tested with the 3-in-1 boards, so that will remain in sof-dev until our testing is enabled.

**Review by RanderWang — DISMISSED** [review]

**Comment by libinyang** [comment]
I also submit a patch which let adl-p use sof-adl-p.ri, not sof-tgl.ri https://github.com/thesofproject/linux/pull/2817

File: sound/soc/intel/common/soc-acpi-intel-adl-match.c [hunk 2]
```diff
@@ -191,6 +200,20 @@ static const struct snd_soc_acpi_link_adr adl_sdca_3_in_1[] = {
...
+static const struct snd_soc_acpi_link_adr adl_sdw_rt1316_link2_rt714_link0[] = {
+	{
+		.mask = BIT(2),
+		.num_adr = ARRAY_SIZE(rt1316_2_single_adr),
+		.adr_d = rt1316_2_single_adr,
+	},
+	{
+		.mask = BIT(0),
+		.num_adr = ARRAY_SIZE(rt714_0_adr),
+		.adr_d = rt714_0_adr,
```
Comment (libinyang) [inline sound/soc/intel/common/soc-acpi-intel-adl-match.c:221]: rt714_0_adr is not defined?

Comment (plbossart) [reply sound/soc/intel/common/soc-acpi-intel-adl-match.c:221]: forgot to push. fixed now.

**Comment by plbossart** [comment]
PR tested by @libinyang, ready for reviews/merge
We will however wait for the next Linux cycle to upstream

**Review by RanderWang — APPROVED** [review]

**Review by libinyang — APPROVED** [review]

**Pull request merged**